src/components/Modals/NewFileModal.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react';
+import { BsFillCheckCircleFill } from 'react-icons/bs';
+import toast from 'react-hot-toast';
+
+import { ConfirmModal } from './index';
+import examples from '../../examples';
+import { EditorService } from '../../services';
+import state from '../../state';
+
+export const NewFileModal: React.FunctionComponent = () => {
+  const sidebarState = state.useSidebarState();
+  const [selectedTemplate, setSelectedTemplate] = useState({ title: '', template: '' });
+
+  const handleTemplateClick = (template: any) => {
+    setSelectedTemplate(template);
+  };
```

Comment:
As I see, you don't make any logic here but only repass the arguments, so please remove that function and start using `setSelectedTemplate` function in places where you used the `handleTemplateClick` function.